lib/auth/helpers.go

```diff
@@ -1340,11 +1340,38 @@ func CreateUser(ctx context.Context, clt clt, username string, roles ...types.Ro
 	return created, trace.Wrap(err)
 }
 
+// createUserAndRoleOptions is a set of options for CreateUserAndRole
+type createUserAndRoleOptions struct {
+	mutateUser func(user types.User)
+	mutateRole func(role types.Role)
+}
+
+// CreateUserAndRoleOption is a functional option for CreateUserAndRole
+type CreateUserAndRoleOption func(*createUserAndRoleOptions)
+
+// WithUserMutator sets a function that will be called to mutate the user before it is created
+func WithUserMutator(mutate func(user types.User)) CreateUserAndRoleOption {
+	return func(o *createUserAndRoleOptions) {
+		o.mutateUser = mutate
```

Comment:
Might be nice to make this composable (use a list of mutateUser funcs?). like if CreateUserWithTraitsAndRole wanted to accept some options and then add its own set of options without overriding a usermutator passed from the caller